server/rest/src/test/java/org/infinispan/rest/assertion/JsonAssertion.java

```diff
@@ -0,0 +1,41 @@
+package org.infinispan.rest.assertion;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNull;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+
+/**
+ * Assertions on JSON contents.
+ *
+ * @author Dan Berindei
+ * @since 12
+ */
+public class JsonAssertion {
+   private Json node;
+
+   public JsonAssertion(Json node) {
+      this.node = node;
+   }
+
+   public JsonAssertion hasProperty(String propertyName) {
+      return new JsonAssertion(node.at(propertyName));
+   }
+
+   public JsonAssertion hasNoProperty(String propertyName) {
+      assertNull(node.at(propertyName));
+      return this;
+   }
+
+   public void is(int value) {
+      assertEquals(value, node.asInteger());
+   }
+
+   public void is(String value) {
+      assertEquals(value, node.asString());
+   }
+
+   public void isNull() {
```

Comment:
This shouldn't be needed if we change `hasProperty` to check for not null.

Reply:
It is needed in order to distinguish `{error: null}` from `{}`.
I just made a typo in the test and instead of fixing that I assumed that the property is parsed as `null` (it's actually a `NullJson`).